### ISSUE TITLE
[BUG] pre-commit is skipping all files in salt/modules/

### ISSUE THREAD

**Issue by danielrobbins** [saltstack/salt#59848]
When using pre-commit, it appears to be skipping all files in salt/modules, as well as some other files, and I cannot find out why.
If you run "black ." in salt/modules you will see that ~89 files need reformatting but for some reason, the current pre-commit config is completely skipping this.

salt/modules does not appear in any exclude lists so this is odd.

**Comment by danielrobbins** [comment]
This appears to be due to a bug in an older version of black that we are using. I have confirmed locally that the following patch fixes this issue:

```
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index c09445a261..5abef397bd 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1118,7 +1118,7 @@ repos:
             )$
 
   - repo: https://github.com/psf/black
-    rev: 19.10b0
+    rev: 20.8b1
     hooks:
       - id: black
         # This tells pre-commit not to pass files to black.
```
With this update, pre-commit --all-files now properly scans salt/modules.

**Comment by danielrobbins** [comment]
The pre-commit run now finds ~450 files that need reformatting. I have confirmed it is still respecting the exclude patterns, but isn't being weird like before and excluding files for no reason. I am wondering if blacking was really functioning at all before. Possibly, the exclude pattern was triggering a bug that resulted in all files being excluded from blacking.

**Comment by s0undt3ch** [comment]
This is not a bug, it's expected behavior.

```
❯ git reset --hard upstream/master
HEAD is now at 2548b92056 Ensure base upgrades in docs venv before run
~/projects/SaltStack/salt/hotfix/black-master (hotfix/black-master↑2629|✔) .venv
❯ black --version
black, version 19.10b0
~/projects/SaltStack/salt/hotfix/black-master (hotfix/black-master↑2629|✔) .venv
❯ black setup.py salt/ tests/
All done! ✨ 🍰 ✨
2963 files left unchanged.
~/projects/SaltStack/salt/hotfix/black-master (hotfix/black-master↑2629|✔) .venv
❯
```

When using the **right** version of black(I'll explain the "right" next), all files in salt are properly blacken.

Now, the right version of black, **currently**, is 19.10b0. That's because at the time we applied black, 20.8b1 wasn't out yet.
Preferably, I would also use this last release as it handles trailing commas way better, but, applying a newer version of black means that almost all, if not all open PRs(660 at the time of writing) need to be rebased, which was a problem on the initial blackening of Salt but we went for it anyway, and it will be a problem again once we start using this new version.

We should only upgrade to the latest black when we decide to completely remove Py2 support from all the code base since it also means rebasing most of the open PRs(if not all).

**Comment by ggiesen** [comment]
Given that #59847 suggests doing exactly that (and I'm supportive of that), perhaps upgrading to the new black should be added to the mix.

**Comment by danielrobbins** [comment]
I was not aware until now of pyupgrade running in the local pre-commit hook, but not running during CI. This forces any files touched by a PR to be magically upgraded to py3, but this change either mixes with the commit history or is a one-off change affecting only a small portion of files, which clutters the commit history and spreads the py3 upgrade over hundreds of commits. This is not great for git history, and I also find the rationale for these rather unusual hoops to be questionable and an overall negative for the project and current and future community contributions. I've gone into a bit more detail in #59847.

**Comment by danielrobbins** [comment]
The original issue was discovered when I created a PR to fix a small issue in salt/modules/quota.py. The GitHub CI failed.

This was not caused by black, as I reported originally. The GitHub CI was expecting the modified file to be py3 compatible, and it was not, because I only committed my tiny changes, and did not run the pre-commit hook, which would have taken the entire file and made it py3 compatible.

I went down a rabbit hole looking at black because I didn't realize that there were two parts of pre-commit that essentially rewrite .py files -- pyupgrade and black. So I'm going to close this issue. Black should be upgraded but that was not the root cause of this problem.
